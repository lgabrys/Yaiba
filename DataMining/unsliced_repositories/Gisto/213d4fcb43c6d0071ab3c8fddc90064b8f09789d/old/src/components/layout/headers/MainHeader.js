import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { baseAppColor, headerColor, lightText } from 'constants/colors';
import { SIDEBAR_WIDTH, logoText } from 'constants/config';
import Icon from 'components/common/Icon';
import UserArea from 'components/UserArea';
import {get} from "lodash/fp";

const HeaderWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  color: #fff;
  vertical-align: middle;
  height: 70px;
  line-height: 11px;
`;

export const HeaderSection = styled.div`
  background: ${baseAppColor};
  flex: 1;
  border-right: 1px solid ${baseAppColor};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  &:last-child {
    border: none;
  }
`;

const Logo = styled(HeaderSection)`
  width: ${SIDEBAR_WIDTH - 40}px;
  max-width: ${SIDEBAR_WIDTH - 40}px;
  padding: 20px;
  a {
    color: ${headerColor};
    text-decoration: none;
  }
`;

const MiddleArea = styled(HeaderSection)``;

const Loading = styled.span`
  align-self: center;
`;

const MainHeader = ({ loading, rateLimit }) => (
  <HeaderWrapper>
    <Logo>
      <Link to="/" title={ `API Rate limit: ${get(['rate', 'remaining'], rateLimit)}/${get(['rate', 'limit'], rateLimit)}` }>{ logoText }</Link>
    </Logo>
    <MiddleArea>
      <Icon color={ lightText } type="menu"/>
      { loading && <Loading><Icon type="loading"/> { 'loading...' }</Loading> }
      <div>{ /* placeholder */ }</div>
    </MiddleArea>
    <UserArea/>
  </HeaderWrapper>
);

const mapStateToProps = (state) => ({
  loading: state.ui.snippets.loading,
  rateLimit: get(['ui', 'rateLimit'], state)
});

MainHeader.propTypes = {
  loading: PropTypes.bool,
  rateLimit: PropTypes.object
};

export default connect(mapStateToProps)(MainHeader);
