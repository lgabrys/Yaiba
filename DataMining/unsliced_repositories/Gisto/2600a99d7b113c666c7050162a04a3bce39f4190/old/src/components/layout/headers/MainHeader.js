import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { HashRouter as Router, Link } from 'react-router-dom';
import styled from 'styled-components';
import { get } from 'lodash/fp';

import * as snippetsActions from 'actions/snippets';

import { SIDEBAR_WIDTH, LOGO_TEXT } from 'constants/config';
import { isEnterpriseLogin } from 'utils/login';
import { toUnixTimeStamp } from 'utils/date';

import UserArea from 'components/AppArea';
import Loading from 'components/common/Loading';
import Button from 'components/common/controls/Button';
import { getSetting } from 'utils/settings';

const HeaderWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  color: #fff;
  vertical-align: middle;
  height: 70px;
  line-height: 11px;
`;

export const HeaderSection = styled.div`
  background: ${(props) => props.theme.baseAppColor};
  flex: 1;
  border-right: 1px solid ${(props) => props.theme.baseAppColor};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  &:last-child {
    border: none;
  }
`;

const Logo = styled(HeaderSection)`
  width: ${SIDEBAR_WIDTH - 20}px;
  max-width: ${SIDEBAR_WIDTH - 20}px;
  padding: 20px 0 20px 20px;
  font-size: 20px;
  a {
    color: ${(props) => props.theme.headerColor};
    text-decoration: none;
  }
`;

const MiddleArea = styled(HeaderSection)``;

const StyledLink = styled(Link)`
  color: ${(props) => props.theme.lightText};
  text-decoration: none;
  line-height: 25px;
`;

const LoadingIndicator = styled.span`
  align-self: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 1;
`;

const RateLimit = styled.span`
  font-size: 12px;
  font-weight: 500;
  text-align: center;
  color: ${(props) => props.theme.headerBgLightest};
  cursor: pointer;

  &:hover {
    color: ${(props) => props.theme.lightText};
  }

  > small {
    font-size: 8px;
  }
`;

export class MainHeader extends Component {
  renderLoading = () => {
    if (!this.props.loading) {
      return <div/>;
    }

    return (
      <LoadingIndicator>
        <Loading text="Loading..."/>
      </LoadingIndicator>
    );
  };

  renderRateLimit = () => {
    const { rateLimit, getRateLimit, settings } = this.props;
    const apiLimitResetTime = Math.floor(
      (get(['rate', 'reset'], rateLimit) - toUnixTimeStamp(new Date().getTime())) / 60
    );

    if (
      !settings['settings-show-api-rate-limit'] ||
      !getSetting('settings-show-api-rate-limit', true)
    ) {
      return null;
    }

    if (get('loading', rateLimit)) {
      return <Loading text=""/>;
    }

    return (
      <RateLimit onClick={ () => getRateLimit() } title="Click to reload">
        <small>API rate limit:</small>
        <br/>
        {`${get(['rate', 'remaining'], rateLimit)}/${get(['rate', 'limit'], rateLimit)}`}
        <br/>
        <small>Reset in {apiLimitResetTime} min.</small>
      </RateLimit>
    );
  };

  render() {
    const { edit, isCreateNew } = this.props;

    return (
      <HeaderWrapper>
        {!edit && (
          <Logo>
            <Link to="/">
              {LOGO_TEXT} {isEnterpriseLogin() && <small>enterprise</small>}
            </Link>
            {!isCreateNew && (
              <Router>
                <Button title="Create new snippet" icon="add" height="30px" outline>
                  <StyledLink to="/new">New snippet</StyledLink>
                </Button>
              </Router>
            )}
          </Logo>
        )}
        <MiddleArea>
          {this.renderLoading()}

          {this.renderRateLimit()}
        </MiddleArea>
        <UserArea/>
      </HeaderWrapper>
    );
  }
}

const mapStateToProps = (state) => ({
  loading: state.ui.snippets.loading,
  rateLimit: get(['ui', 'rateLimit'], state),
  edit: get(['ui', 'snippets', 'edit'], state),
  settings: get(['ui', 'settings'], state),
  isCreateNew: get(['router', 'location', 'pathname'], state) === '/new'
});

MainHeader.propTypes = {
  loading: PropTypes.bool,
  edit: PropTypes.bool,
  rateLimit: PropTypes.object,
  settings: PropTypes.object,
  isCreateNew: PropTypes.bool,
  getRateLimit: PropTypes.func
};

export default connect(
  mapStateToProps,
  {
    getRateLimit: snippetsActions.getRateLimit
  }
)(MainHeader);
