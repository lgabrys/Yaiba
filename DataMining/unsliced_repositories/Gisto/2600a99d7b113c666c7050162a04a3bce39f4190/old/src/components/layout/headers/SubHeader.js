import React from 'react';
import PropTypes from 'prop-types';
import { get } from 'lodash/fp';
import { connect } from 'react-redux';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import styled from 'styled-components';

import Search from 'components/Search';
import SnippetHeader from 'components/layout/headers/SnippetHeader';

const SubHeaderWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  vertical-align: middle;
  height: 60px;
  box-shadow: 0 1px 2px ${(props) => props.theme.boxShadow};
  z-index: 2;
`;

const SnippetHeaderBlock = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  flex: 1;
  align-items: center;
  margin: 0 0 0 10px;
`;

export const SubHeader = ({ edit, isCreateNew }) => (
  <SubHeaderWrapper>
    {!edit && !isCreateNew && <Search/>}
    <SnippetHeaderBlock>
      <Router>
        <Switch>
          <Route exact path="/" component={ () => <div>Dashboard</div> }/>
          <Route exact path="/about" component={ () => <div>About</div> }/>
          <Route exact path="/settings" component={ () => <div>Settings</div> }/>
          <Route path="/snippet/:id" component={ SnippetHeader }/>
        </Switch>
      </Router>
    </SnippetHeaderBlock>
  </SubHeaderWrapper>
);

const mapStateToProps = (state) => ({
  edit: get(['ui', 'snippets', 'edit'], state),
  isCreateNew: get(['router', 'location', 'pathname'], state) === '/new'
});

SubHeader.propTypes = {
  edit: PropTypes.bool,
  isCreateNew: PropTypes.bool
};

export default connect(mapStateToProps)(SubHeader);
