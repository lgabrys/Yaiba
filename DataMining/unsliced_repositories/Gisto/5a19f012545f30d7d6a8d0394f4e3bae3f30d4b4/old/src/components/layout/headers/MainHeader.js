import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { baseAppColor, headerColor, lightText } from 'constants/colors';
import { SIDEBAR_WIDTH, logoText } from 'constants/config';
import Icon from 'components/common/Icon';
import UserArea from 'components/AppArea';
import { get } from 'lodash/fp';

const HeaderWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  color: #fff;
  vertical-align: middle;
  height: 70px;
  line-height: 11px;
`;

export const HeaderSection = styled.div`
  background: ${baseAppColor};
  flex: 1;
  border-right: 1px solid ${baseAppColor};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  &:last-child {
    border: none;
  }
`;

const Logo = styled(HeaderSection)`
  width: ${SIDEBAR_WIDTH - 40}px;
  max-width: ${SIDEBAR_WIDTH - 40}px;
  padding: 20px;
  font-size: 20px;
  a {
    color: ${headerColor};
    text-decoration: none;
  }
`;

const MiddleArea = styled(HeaderSection)``;

const Loading = styled.span`
  align-self: center;
`;

export const MainHeader = ({ loading, rateLimit, edit }) => (
  <HeaderWrapper>
    { !edit && (
      <Logo>
        <Link to="/" title={ `API Rate limit: ${get(['rate', 'remaining'], rateLimit)}/${get(['rate', 'limit'], rateLimit)}` }>{ logoText }</Link>
      </Logo>
    ) }
    <MiddleArea>
      <Icon color={ lightText } type="menu"/>
      { loading && <Loading><Icon type="loading"/> { 'loading...' }</Loading> }
      <div>{ /* placeholder */ }</div>
    </MiddleArea>
    <UserArea/>
  </HeaderWrapper>
);

const mapStateToProps = (state) => ({
  loading: state.ui.snippets.loading,
  rateLimit: get(['ui', 'rateLimit'], state),
  edit: get(['ui', 'snippets', 'edit'], state)
});

MainHeader.propTypes = {
  loading: PropTypes.bool,
  edit: PropTypes.bool,
  rateLimit: PropTypes.object
};

export default connect(mapStateToProps)(MainHeader);
