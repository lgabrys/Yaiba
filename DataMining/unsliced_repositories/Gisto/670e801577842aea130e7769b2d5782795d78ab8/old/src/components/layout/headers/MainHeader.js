import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { HashRouter as Router, Link } from 'react-router-dom';
import styled from 'styled-components';
import { get } from 'lodash/fp';

import { baseAppColor, headerColor, lightText } from 'constants/colors';
import { SIDEBAR_WIDTH, logoText } from 'constants/config';
import { isEnterpriseLogin } from 'utils/login';

import UserArea from 'components/AppArea';
import Loading from 'components/common/Loading';
import Button from 'components/common/controls/Button';

const HeaderWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  color: #fff;
  vertical-align: middle;
  height: 70px;
  line-height: 11px;
`;

export const HeaderSection = styled.div`
  background: ${baseAppColor};
  flex: 1;
  border-right: 1px solid ${baseAppColor};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  &:last-child {
    border: none;
  }
`;

const Logo = styled(HeaderSection)`
  width: ${SIDEBAR_WIDTH - 40}px;
  max-width: ${SIDEBAR_WIDTH - 40}px;
  padding: 20px;
  font-size: 20px;
  a {
    color: ${headerColor};
    text-decoration: none;
  }
`;

const MiddleArea = styled(HeaderSection)``;

const StyledLink = styled(Link)`
  color: ${lightText};
  text-decoration: none;
  line-height: 25px;
`;

const LoadingIndicator = styled.span`
  align-self: center;
`;

export const MainHeader = ({ loading, rateLimit, edit }) => (
  <HeaderWrapper>
    { !edit && (
      <Logo>
        <Link to="/" title={ `API Rate limit: ${get(['rate', 'remaining'], rateLimit)}/${get(['rate', 'limit'], rateLimit)}` }>
          { logoText } { isEnterpriseLogin() && <small>enterprise</small> }
        </Link>
      </Logo>
    ) }
    <MiddleArea>
      { !edit && (
        <Router>
          <Button icon="add" height="30px" outline>
            <StyledLink to="/new">New snippet</StyledLink>
          </Button>
        </Router>
      ) }
      { loading && (
        <LoadingIndicator>
          <Loading text="Loading..."/>
        </LoadingIndicator>
      ) }
      <div>{ /* placeholder */ }</div>
    </MiddleArea>
    <UserArea/>
  </HeaderWrapper>
);

const mapStateToProps = (state) => ({
  loading: state.ui.snippets.loading,
  rateLimit: get(['ui', 'rateLimit'], state),
  edit: get(['ui', 'snippets', 'edit'], state)
});

MainHeader.propTypes = {
  loading: PropTypes.bool,
  edit: PropTypes.bool,
  rateLimit: PropTypes.object
};

export default connect(mapStateToProps)(MainHeader);
