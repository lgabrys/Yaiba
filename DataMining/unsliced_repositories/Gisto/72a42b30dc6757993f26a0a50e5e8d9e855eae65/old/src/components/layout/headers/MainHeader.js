import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { HashRouter as Router, Link } from 'react-router-dom';
import styled from 'styled-components';
import { get } from 'lodash/fp';

import * as snippetsActions from 'actions/snippets';

import {
  baseAppColor, headerBgLightest, headerColor, lightText
} from 'constants/colors';
import { SIDEBAR_WIDTH, logoText } from 'constants/config';
import { isEnterpriseLogin } from 'utils/login';
import { toUnixTimeStamp } from 'utils/date';

import UserArea from 'components/AppArea';
import Loading from 'components/common/Loading';
import Button from 'components/common/controls/Button';

const HeaderWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  color: #fff;
  vertical-align: middle;
  height: 70px;
  line-height: 11px;
`;

export const HeaderSection = styled.div`
  background: ${baseAppColor};
  flex: 1;
  border-right: 1px solid ${baseAppColor};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  &:last-child {
    border: none;
  }
`;

const Logo = styled(HeaderSection)`
  width: ${SIDEBAR_WIDTH - 20}px;
  max-width: ${SIDEBAR_WIDTH - 20}px;
  padding: 20px 0 20px 20px;
  font-size: 20px;
  a {
    color: ${headerColor};
    text-decoration: none;
  }
`;

const MiddleArea = styled(HeaderSection)``;

const StyledLink = styled(Link)`
  color: ${lightText};
  text-decoration: none;
  line-height: 25px;
`;

const LoadingIndicator = styled.span`
  align-self: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 1;
`;

const RateLimit = styled.span`
  font-size: 12px;
  font-weight: 500;
  text-align: center;
  color: ${headerBgLightest};
  cursor: pointer;
  
  &:hover {
    color: ${lightText};
  }
  
  > small {
    font-size: 8px;
  } 
`;

export const MainHeader = ({
  loading, rateLimit, edit, isCreateNew, getRateLimit
}) => {
  const apiLimitResetTime = Math.floor((get(['rate', 'reset'], rateLimit) - toUnixTimeStamp(new Date().getTime())) / 60);

  return (
    <HeaderWrapper>
      {!edit && (
        <Logo>
          <Link to="/">
            {logoText} {isEnterpriseLogin() && <small>enterprise</small>}
          </Link>
          {!isCreateNew && (
            <Router>
              <Button icon="add" height="30px" outline>
                <StyledLink to="/new">New snippet</StyledLink>
              </Button>
            </Router>
          )}
        </Logo>
      )}
      <MiddleArea>
        {loading ? (
          <LoadingIndicator>
            <Loading text="Loading..."/>
          </LoadingIndicator>
        ) : <div/>}

        { get(['loading'], rateLimit) ? (
          <Loading text=""/>
        ) : (
          <RateLimit onClick={ () => getRateLimit() }
                     title="Click to reload">
            <small>API rate limit:</small>
            <br/>
            {`${get(['rate', 'remaining'], rateLimit)}/${get(['rate', 'limit'], rateLimit)}`}
            <br/>
            <small>Reset in { apiLimitResetTime } min.</small>
          </RateLimit>
        ) }

      </MiddleArea>
      <UserArea/>
    </HeaderWrapper>
  );
};

const mapStateToProps = (state) => ({
  loading: state.ui.snippets.loading,
  rateLimit: get(['ui', 'rateLimit'], state),
  edit: get(['ui', 'snippets', 'edit'], state),
  isCreateNew: get(['router', 'location', 'pathname'], state) === '/new'
});

MainHeader.propTypes = {
  loading: PropTypes.bool,
  edit: PropTypes.bool,
  rateLimit: PropTypes.object,
  isCreateNew: PropTypes.bool,
  getRateLimit: PropTypes.func
};

export default connect(mapStateToProps, {
  getRateLimit: snippetsActions.getRateLimit
})(MainHeader);
