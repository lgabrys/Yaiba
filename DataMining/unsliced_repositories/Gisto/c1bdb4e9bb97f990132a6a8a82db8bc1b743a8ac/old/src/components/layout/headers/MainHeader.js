import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { HashRouter as Router, Link } from 'react-router-dom';
import styled from 'styled-components';
import { get } from 'lodash/fp';

import * as snippetsActions from 'actions/snippets';

import {
  baseAppColor, headerBgLightest, headerColor, lightText
} from 'constants/colors';
import { SIDEBAR_WIDTH, logoText } from 'constants/config';
import { isEnterpriseLogin } from 'utils/login';
import { toUnixTimeStamp } from 'utils/date';

import UserArea from 'components/AppArea';
import Loading from 'components/common/Loading';
import Button from 'components/common/controls/Button';
import { getSetting } from 'utils/settings';

const HeaderWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  color: #fff;
  vertical-align: middle;
  height: 70px;
  line-height: 11px;
`;

export const HeaderSection = styled.div`
  background: ${baseAppColor};
  flex: 1;
  border-right: 1px solid ${baseAppColor};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  &:last-child {
    border: none;
  }
`;

const Logo = styled(HeaderSection)`
  width: ${SIDEBAR_WIDTH - 20}px;
  max-width: ${SIDEBAR_WIDTH - 20}px;
  padding: 20px 0 20px 20px;
  font-size: 20px;
  a {
    color: ${headerColor};
    text-decoration: none;
  }
`;

const MiddleArea = styled(HeaderSection)``;

const StyledLink = styled(Link)`
  color: ${lightText};
  text-decoration: none;
  line-height: 25px;
`;

const LoadingIndicator = styled.span`
  align-self: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 1;
`;

const RateLimit = styled.span`
  font-size: 12px;
  font-weight: 500;
  text-align: center;
  color: ${headerBgLightest};
  cursor: pointer;
  
  &:hover {
    color: ${lightText};
  }
  
  > small {
    font-size: 8px;
  } 
`;

export class MainHeader extends Component {
  renderLoading = () => {
    if (!this.props.loading) {
      return <div/>;
    }

    return (
      <LoadingIndicator>
        <Loading text="Loading..."/>
      </LoadingIndicator>
    );
  };

  renderRateLimit = () => {
    const { rateLimit, getRateLimit } = this.props;
    const apiLimitResetTime = Math.floor((get(['rate', 'reset'], rateLimit) - toUnixTimeStamp(new Date().getTime())) / 60);

    if (!getSetting('settings-show-api-rate-limit', true)) {
      return null;
    }

    if (get('loading', rateLimit)) {
      return (
        <Loading text=""/>
      );
    }

    return (
      <RateLimit onClick={ () => getRateLimit() }
                 title="Click to reload">
        <small>API rate limit:</small>
        <br/>
        { `${get(['rate', 'remaining'], rateLimit)}/${get(['rate', 'limit'], rateLimit)}` }
        <br/>
        <small>Reset in { apiLimitResetTime } min.</small>
      </RateLimit>
    );
  };

  render() {
    const { edit, isCreateNew } = this.props;

    return (
      <HeaderWrapper>
        { !edit && (
          <Logo>
            <Link to="/">
              { logoText } { isEnterpriseLogin() && <small>enterprise</small> }
            </Link>
            { !isCreateNew && (
              <Router>
                <Button icon="add" height="30px" outline>
                  <StyledLink to="/new">New snippet</StyledLink>
                </Button>
              </Router>
            ) }
          </Logo>
        ) }
        <MiddleArea>
          { this.renderLoading() }

          { this.renderRateLimit() }

        </MiddleArea>
        <UserArea/>
      </HeaderWrapper>
    );
  }
}

const mapStateToProps = (state) => ({
  loading: state.ui.snippets.loading,
  rateLimit: get(['ui', 'rateLimit'], state),
  edit: get(['ui', 'snippets', 'edit'], state),
  isCreateNew: get(['router', 'location', 'pathname'], state) === '/new'
});

MainHeader.propTypes = {
  loading: PropTypes.bool,
  edit: PropTypes.bool,
  rateLimit: PropTypes.object,
  isCreateNew: PropTypes.bool,
  getRateLimit: PropTypes.func
};

export default connect(mapStateToProps, {
  getRateLimit: snippetsActions.getRateLimit
})(MainHeader);
