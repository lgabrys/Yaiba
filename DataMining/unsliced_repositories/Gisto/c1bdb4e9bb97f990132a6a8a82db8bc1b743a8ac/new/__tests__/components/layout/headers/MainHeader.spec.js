import { MainHeader } from 'components/layout/headers/MainHeader';

const propSetup = (props = {}) => ({
  settings: {
    'settings-show-api-rate-limit': true
  },
  ...props
});

const setup = (props) => shallow(<MainHeader { ...propSetup(props) }/>);

describe('COMPONENTS - <MainHeader>', () => {
  test('render MainHeader', () => {
    const component = setup({
      loading: false,
      edit: false,
      rateLimit: {
        remaining: 5,
        limit: 10
      }
    });

    expect(component).toMatchSnapshot();
  });
});
