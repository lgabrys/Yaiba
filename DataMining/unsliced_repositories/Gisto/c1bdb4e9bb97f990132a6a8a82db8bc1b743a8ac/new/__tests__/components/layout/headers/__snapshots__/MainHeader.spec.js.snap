// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS - <MainHeader> render MainHeader 1`] = `
<ForwardRef>
  <ForwardRef>
    <Link
      replace={false}
      to="/"
    >
      { Gisto }
       
    </Link>
    <HashRouter>
      <Connect(Button)
        height="30px"
        icon="add"
        outline={true}
      >
        <ForwardRef
          to="/new"
        >
          New snippet
        </ForwardRef>
      </Connect(Button)>
    </HashRouter>
  </ForwardRef>
  <ForwardRef>
    <div />
    <ForwardRef
      onClick={[Function]}
      title="Click to reload"
    >
      <small>
        API rate limit:
      </small>
      <br />
      undefined/undefined
      <br />
      <small>
        Reset in 
        NaN
         min.
      </small>
    </ForwardRef>
  </ForwardRef>
  <ForwardRef />
</ForwardRef>
`;
